nemo/collections/tts/modules/aligner.py

```diff
@@ -143,18 +143,21 @@ def get_mean_distance_for_word(l2_dists, durs, start_token, num_tokens):
 
         return dist_sum / total_frames
 
-    def forward(self, queries, keys, mask=None, attn_prior=None):
+    def forward(self, queries, keys, mask=None, attn_prior=None, conditioning=None):
         """Forward pass of the aligner encoder.
 
         Args:
             queries (torch.tensor): B x C x T1 tensor (probably going to be mel data).
             keys (torch.tensor): B x C2 x T2 tensor (text data).
             mask (torch.tensor): B x T2 x 1 tensor, binary mask for variable length entries (True = mask element, False = leave unchanged).
             attn_prior (torch.tensor): prior for attention matrix.
+            conditioning (torch.tensor): B x T2 x 1 conditioning embedding
         Output:
             attn (torch.tensor): B x 1 x T1 x T2 attention mask. Final dim T2 should sum to 1.
             attn_logprob (torch.tensor): B x 1 x T1 x T2 log-prob attention mask.
         """
+        if conditioning is not None:
+            keys = keys + conditioning.transpose(1,2)
```

Comment:
Out of curiosity, what is the benefit of passing the "conditioning" vector as an argument to the aligner, as opposed to having the method which calls aligner modify the keys?

Right now the FP module is conditioning the input for the pitch/duration predictors while the encoder & aligner take is as an input, which looks like an inconsistent design pattern.

Reply:
I think transformers in our implementation has the conditioning parameter and other CNN/DNNs do not. We can change that for sure, but it should be a separate refactoring PR.